packages/utils/src/ClipboardUtils.test.ts

```diff
@@ -0,0 +1,49 @@
+import { copyToClipboard } from './ClipboardUtils';
+
+document.execCommand = jest.fn();
+
+describe('Clipboard', () => {
+  describe('writeText', () => {
+    it('should call clipboard.writeText', async () => {
+      Object.assign(navigator, {
+        clipboard: {
+          // eslint-disable-next-line @typescript-eslint/no-empty-function
+          writeText: () => {},
+        },
+      });
+      jest.spyOn(navigator.clipboard, 'writeText');
+
+      await copyToClipboard('test');
+      expect(navigator.clipboard.writeText).toHaveBeenCalledWith('test');
+    });
+
+    it('should throw an error when writeText fails and call copyToClipboardExecCommand', async () => {
+      Object.assign(navigator, {
+        clipboard: {
+          writeText: () => {
+            throw new Error('Could not write text');
+          },
+        },
+      });
+      jest.spyOn(navigator.clipboard, 'writeText');
+
+      await expect(copyToClipboard('test')).rejects.toThrowError(
+        'Unable to execute copy command'
+      );
+      expect(navigator.clipboard.writeText).toThrowError(
```

Comment:
NBD, but this tests the mock defined in the same `it` block. Ok to leave as is if we just want to make sure the mock is functional.

Reply:
I wanted to test the case where `writeText` throws. Should the mock be defined outside?

Reply:
No need to define the mock outside. What I meant in my comment - this line `expect(navigator.clipboard.writeText).toThrowError...` would pass even without calling `copyToClipboard('test')` above:

```
it('passes without calling copyToClipboard ', async () => {
      Object.assign(navigator, {
        clipboard: {
          writeText: () => {
            throw new Error('Could not write text');
          },
        },
      });
      expect(navigator.clipboard.writeText).toThrowError(
        'Could not write text'
      );
    });
```


Again, no big deal, just wanted to point out that `expect(navigator.clipboard.writeText)` tests the mock and not the `copyToClipboard` function.